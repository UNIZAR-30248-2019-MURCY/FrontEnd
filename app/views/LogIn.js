import React, {Component} from 'react';
import {
    View,
    TextInput,
    StyleSheet,
    Image,
    Text
} from 'react-native'
import {Button} from 'react-native-elements';
import Icon from 'react-native-vector-icons/FontAwesome';
import {logInUser, signUpUser} from "../services/user/userFuncs";

export default class LogIn extends Component {

    constructor(props) {
        super(props);
        this.state = {
            email: '',
            password: '',
            error: false
        }
        this.onChangeText = this.onChangeText.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    onChangeText = (key, val) => {
        this.setState({[key]: val})
    }

    handleSubmit() {
        logInUser(this.state.email, this.state.password)
            .then((data) => {
                console.log(data);
                //this.setState(data);
            })
            .catch((error) => {
                console.log("Api call error");
                alert(error.message);
            });
    }

    render() {
        return (
            <View style={styles.container}>
                <View style={styles.cross}>
                    <Button
                        type="clear"
                        icon={
                            <Icon
                                name="times"
                                size={30}
                                color="grey"
                            />
                        }
                        onPress={() => {
                            this.props.history.goBack();
                        }
                        }
                    />
                </View>

                <View style={styles.login}>
                <Image
                    style={styles.logo}
                    source={require('./../assets/images/murcy.png')}
                />
                <TextInput
                    style={styles.input}
                    placeholder='Email'
                    autoCapitalize="none"
                    placeholderTextColor='darkgrey'
                    onChangeText={val => this.onChangeText('email', val)}
                />
                <TextInput
                    style={styles.input}
                    placeholder='Password'
                    secureTextEntry={true}
                    autoCapitalize="none"
                    placeholderTextColor='darkgrey'
                    onChangeText={val => this.onChangeText('password', val)}
                />

                <Button
                    buttonStyle={styles.button}
                    title="Log In"
                    onPress={() => {
                        this.handleSubmit()
                    }}/>

                <Button
                    className='signup-button'
                    type="clear"
                    buttonStyle={styles.button2}
                    title="Sign Up"
                    titleStyle={{color: 'grey'}}
                    onPress={() => {
                        this.props.history.replace('/signup');
                    }
                    }/>

                </View>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        //justifyContent: 'center',
    },
    cross: {
        marginTop: 50,
        marginBottom: 40,
        marginRight: 10,
        alignItems: 'flex-end',
    },
    login: {
        flex: 1,
        alignItems: 'center',
        //justifyContent: 'center',
    },
    logo: {
        marginBottom: 30,
        padding: 8,
        width: 100,
        height: 100
    },
    input: {
        width: 300,
        height: 55,
        backgroundColor: 'whitesmoke',
        margin: 10,
        padding: 8,
        color: 'grey',
        borderRadius: 14,
        fontSize: 18,
        fontWeight: '500',
    },
    button: {
        width: 150,
        height: 55,
        marginTop: 30,
        margin: 10,
        backgroundColor: 'grey',
        borderRadius: 14
    },
    button2: {
        width: 150,
        height: 55,
        borderRadius: 14
    }

})
