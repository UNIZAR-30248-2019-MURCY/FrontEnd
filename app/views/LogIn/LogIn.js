import React, {Component} from 'react';
import {
    View,
    TextInput,
    StyleSheet,
    Image,
    Text,
    ActivityIndicator
} from 'react-native'
import {Button} from 'react-native-elements';
import Icon from 'react-native-vector-icons/FontAwesome';
import {logInUser, userInfo} from "../../services/user/userFuncs";
import {saveData} from "../../services/AsyncStorage/save";
import {NavigationActions, StackActions} from "react-navigation";


export default class LogIn extends Component {

    constructor(props) {
        super(props);
        this.state = {
            username: '',
            password: '',
            token: '',
            error: false,
            userData: '',
            loading: false
        }
        this.onChangeText = this.onChangeText.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    onChangeText = (key, val) => {
        this.setState({[key]: val})
    }

    handleSubmit() {
        if (this.state.username !== '' && this.state.password !== '') {
            this.setState({loading: true})
            this.setState({error: false})
            logInUser(this.state.username, this.state.password)
                .then((data) => {
                    saveData('token', data).then(r =>
                        userInfo(data.jsonWebToken).then((dataUser) => {
                                saveData('role', dataUser.role).then(r =>
                                        console.log(dataUser.role),
                                    this.setState({loading: false}),
                                    this.props.navigation.navigate('App')
                                )
                            }
                        )
                            .catch((error) => {
                                this.setState({error: error.message}),
                                    this.setState({loading: false})
                            })
                    );
                })
                .catch((error) => {
                    this.setState({error: error.message}),
                        this.setState({loading: false})
                });
        } else {
            this.setState({error: 'Introduzca todos los campos'})
        }
    }

    render() {
        let showErr = (
            this.state.error ?
                <View style={styles.error} className='errorShow'>
                    <Text style={{color: 'red'}}>
                        {this.state.error}
                    </Text>
                </View> :
                <View></View>
        );
        let showLoading = (
            this.state.loading ?
                <View style={[styles.containerLoading]}>
                    <ActivityIndicator animating={this.state.loading} size="large" color="grey"/>
                </View> :
                <View></View>
        );

        return (

            <View style={styles.container}>
                <View style={styles.cross}>
                    <Button
                        className='close-button'
                        type="clear"
                        icon={
                            <Icon
                                name="times"
                                size={30}
                                color="grey"
                            />
                        }
                        onPress={() => {
                            this.props.navigation.navigate('AuthLoading')
                        }
                        }
                    />
                </View>

                <View style={styles.login}>
                    <Image
                        style={styles.logo}
                        source={require('../../assets/images/murcy.png')}
                    />
                    <TextInput
                        className='userInput'
                        style={styles.input}
                        placeholder='Username'
                        autoCapitalize="none"
                        placeholderTextColor='darkgrey'
                        onChangeText={val => this.onChangeText('username', val)}
                    />
                    <TextInput
                        className='passInput'
                        style={styles.input}
                        placeholder='Password'
                        secureTextEntry={true}
                        autoCapitalize="none"
                        placeholderTextColor='darkgrey'
                        onSubmitEditing={() => {
                            this.handleSubmit()
                        }}
                        onChangeText={val => this.onChangeText('password', val)}
                    />
                    {showErr}
                    {showLoading}
                    <Button
                        className='login-button'
                        buttonStyle={styles.button}
                        title="Log In"
                        onPress={() => {
                            this.handleSubmit()
                        }}/>
                    <Button
                        className='signup-button'
                        type="clear"
                        buttonStyle={styles.button2}
                        title="Sign Up"
                        titleStyle={{color: 'grey'}}
                        onPress={() => {
                            this.props.navigation.replace('SignUp');
                        }
                        }/>
                </View>

            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    containerLoading: {
        flexDirection: 'row',
        justifyContent: 'space-around',
        padding: 10
    },
    cross: {
        marginTop: 50,
        marginBottom: 40,
        marginRight: 10,
        alignItems: 'flex-end',
    },
    login: {
        alignItems: 'center',
    },
    logo: {
        marginBottom: 30,
        padding: 8,
        width: 100,
        height: 100
    },
    input: {
        width: 300,
        height: 55,
        backgroundColor: 'whitesmoke',
        margin: 10,
        padding: 8,
        color: 'grey',
        borderRadius: 14,
        fontSize: 18,
        fontWeight: '500',
    },
    button: {
        width: 150,
        height: 55,
        marginTop: 20,
        margin: 10,
        backgroundColor: 'grey',
        borderRadius: 14
    },
    button2: {
        width: 150,
        height: 55,
        borderRadius: 14
    },
    error: {
        margin: 10,

    },

})
