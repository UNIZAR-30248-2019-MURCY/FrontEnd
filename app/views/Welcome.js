import React, {Component} from 'react';
import {
    View,
    StyleSheet,
    Image, ActivityIndicator,
    Platform, Linking
} from 'react-native'
import {Button, colors, Text} from 'react-native-elements';
import {retrieveItem} from "../modules/AsyncStorage/retrieve";
import {emailVerif, getRequestEdit} from "../services/user/userFuncs";
import Icon from 'react-native-vector-icons/FontAwesome';

export default class Welcome extends Component {

    constructor(props) {
        super(props);
        this.state = {
            loading: true
        }
    }
    componentDidMount() {
        if (this.props.navigation) {
            if(this.props.navigation.getParam('token')){
                this.props.navigation.navigate('EmailVerif', {
                    token: this.props.navigation.getParam('token')
                });
            }
        }
        this._bootstrapAsync();
    }


    _bootstrapAsync = async () => {
        retrieveItem('token')
            .then(data => {
                if (data) {
                    this.setState({loading: false})
                    this.props.navigation.navigate('App');
                } else {
                    this.setState({loading: false})
                }

            })
            .catch((error) => {
                this.setState({loading: false})
            })
    };

    render() {
        let showLoading = (
            this.state.loading ?
                <View style={styles.horizontal}>
                    <ActivityIndicator animating={this.state.loading} size="large" color="grey"/>
                </View>
                :
                <Button
                    className='enter-button'
                    type="clear"
                    buttonStyle={styles.button}
                    title="Enter"
                    titleStyle={styles.buttonText}
                    onPress={() => {
                        this.props.navigation.navigate('LogIn');
                    }}/>

        );
        let welcomePlatform = (
            Platform.OS === 'web' ?
                <View style={styles.container}>
                    <View style={styles.containerMurcyWeb}>
                        <Image
                            style={styles.logo}
                            source={require('./../assets/images/murcy.png')}
                        />
                        <Text h1>Murcy</Text>
                        {showLoading}
                    </View>
                    <View style={styles.containerDownload}>
                        <Text style={styles.text}>Download our App!</Text>
                        <View style={styles.containerPlatforms}>
                            <View style={styles.itemIcon}>
                                <Button
                                    className='close-button'
                                    type="clear"
                                    icon={
                                        <Icon
                                            name="apple"
                                            size={25}
                                            color="grey"
                                        />
                                    }
                                    onPress={() => {
                                        Linking.openURL('https://expo.io/@javier_mixture/murcy')
                                    }
                                    }
                                />
                            </View>
                            <View style={styles.itemIcon}>
                                <Button
                                    className='close-button'
                                    type="clear"
                                    icon={
                                        <Icon
                                            name="android"
                                            size={25}
                                            color="grey"
                                        />
                                    }
                                    onPress={() => {
                                        Linking.openURL('https://expo.io/@javier_mixture/murcy')
                                    }
                                    }
                                />
                            </View>
                        </View>
                    </View>
                </View>
                :
                <View style={styles.container}>
                    <View style={styles.containerMurcy}>
                        <Image
                            style={styles.logo}
                            source={require('./../assets/images/murcy.png')}
                        />
                        <Text h1>Murcy</Text>
                        {showLoading}
                    </View>
                </View>
        );

        return (
            welcomePlatform
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        //justifyContent: 'center',
        padding: 20
    },
    containerMurcy: {
        flex: 1,
        alignItems: 'center',
        //justifyContent: 'center',
        marginTop: 200,
        padding: 20
    },
    containerMurcyWeb: {
        flex: 1,
        alignItems: 'center',
        //justifyContent: 'center',
        marginTop: 100,
        padding: 20
    },
    horizontal: {
        flexDirection: 'row',
        justifyContent: 'space-around',
        padding: 10,
        marginTop: 50,
    },
    logo: {
        margin: 30,
        marginBottom: 15,
        padding: 8,
        width: 100,
        height: 100
    },
    button: {
        marginTop: 30,
        width: 150,
        height: 55,
        padding: 8,
    },
    buttonText: {
        fontSize: 25,
        color: 'grey'
    },
    containerDownload: {
        flex: 1,
        alignItems: 'center',
        //justifyContent: 'center',
        marginTop: 95,
        padding: 20
    },
    downButton: {
        width: 250,
        height: 55,
        padding: 8,
    },
    text: {
        fontSize: 18,
        color: 'grey'
    },
    containerPlatforms: {
        flex: 1,
        flexDirection: 'row',
        flexWrap: 'wrap',
        alignItems: 'flex-start' // if you want to fill rows left to right
    },
    itemIcon: {
        width: '50%' // is 50% of container width
    }

})
