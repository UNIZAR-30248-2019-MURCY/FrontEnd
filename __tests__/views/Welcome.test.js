import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Welcome from '../../app/views/Welcome';
import LogIn from "../../app/views/LogIn";

describe('<Welcome />', () => {
    it('Welcome renders without crashing', () => {
        const rendered = renderer.create(<Welcome />).toJSON();
        expect(rendered).toBeTruthy();
    });
    it('Welcome test against snapshot', () => {
        const welcomePage = renderer.create(<Welcome />).toJSON();
        expect(welcomePage).toMatchSnapshot();
    });
    it('should navigate to Log In', () => {
        const historyMock = { push: jest.fn() };
        const wrapper = shallow(<Welcome history={historyMock} />);
        wrapper.find('.enter-button').simulate('press');
        expect(historyMock.push.mock.calls.length).toEqual(1);
        expect(historyMock.push.mock.calls[0]).toEqual(['/login']);
    })
});
