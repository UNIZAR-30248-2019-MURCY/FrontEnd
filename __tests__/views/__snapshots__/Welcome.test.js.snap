// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Welcome /> Welcome test against snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginTop": 160,
        "padding": 20,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 100,
          "margin": 30,
          "marginBottom": 15,
          "padding": 8,
          "width": 100,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 50,
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Murcy
    </Text>
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 55,
              "justifyContent": "center",
              "marginTop": 70,
              "padding": 8,
              "width": 150,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "grey",
                "fontSize": 25,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Entrar
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
